source/index.js

```diff
@@ -6,20 +6,99 @@ var Renderer = require("./scripts/Renderer")
 
 var WIDTH = 854, HEIGHT = 480
 
+class Gardener {
+    constructor() {
+        this.width = 32
+        this.height = 32
+
+        this.position = {x: 64, y: 64}
+        this.velocity = {x: 0, y: 0}
+
+        this.speed = 0.25 //pixels per millisecond
+    }
+    update(tick) {
+        if(Input.isDown("W")
+        || Input.isDown("<up>")) {
+            this.velocity.y = -1 * this.speed
+        }
+        if(Input.isDown("S")
+        || Input.isDown("<down>")) {
+            this.velocity.y = +1 * this.speed
+        }
+        if(Input.isDown("A")
+        || Input.isDown("<left>")) {
+            this.velocity.x = -1 * this.speed
+        }
+        if(Input.isDown("D")
+        || Input.isDown("<right>")) {
+            this.velocity.x = +1 * this.speed
+        }
+
+        this.position.x += this.velocity.x * tick
+        this.position.y += this.velocity.y * tick
```

Comment:
@mcfunkypants, right here, I should be doing some MATH to get a vector of my velocity, and then translate along that vector so I don't move faster diagonally than I do orthogonally. But I just can't figure out how to do that right now, so I'm skipping it.